src/test/java/uk/ac/ebi/embl/api/validation/check/entry/SubmitterAccessionCheckTest.java

```diff
@@ -109,25 +104,54 @@ public void testCheck_OverMaximumLengthSubmitterAccession() throws Exception {
     }
   }
 
-  /** Test that check fails if the submitter accession is over maximum length. */
+  /** Test that check doesn't fail if the submitter accession is over maximum length and ignoreErrors = true. */
   @Test
   public void testCheck_OverMaximumLengthSubmitterAccessionAndIgnoreError() throws Exception {
     for (ValidationScope validationScope : ValidationScope.values()) {
       TestValidationPlan validationPlan = testValidationPlan(validationScope);
       EntryFactory entryFactory = new EntryFactory();
       Entry entry = entryFactory.createEntry();
       entry.setSubmitterAccession("012345678901234567890123456789012345678901234567891");
-      validationPlan.getProperty().getOptions().ignoreError = Optional.of(true);
+      validationPlan.getProperty().getOptions().ignoreErrors = true;
 
       ValidationResult result = validationPlan.execute(entry);
-      if (validationScope == ValidationScope.ASSEMBLY_CHROMOSOME
-          || validationScope == ValidationScope.ASSEMBLY_SCAFFOLD
-          || validationScope == ValidationScope.ASSEMBLY_CONTIG
-          || validationScope == ValidationScope.ASSEMBLY_TRANSCRIPTOME) {
+      if (isValidScopeForThisTest(validationScope)) {
         assertTrue(result.isValid());
-      } else {
+      } 
+    }
+  }
+
+
+  /** Test that check doesn't fail if the submitter accession is over maximum length and 
+   * assembly type = AssemblyType.BINNEDMETAGENOME. */
```

Comment:
- Proposed checks do not check all assembly types and set assembly type regardless of ValidationScope that is very confusing
- Merge tests as they are identical except for exceptError and input
```
boolean expectError = ...
for (ValidationScope validationScope : ValidationScope.values()) {
   ....
   if (validationScope.group == Group.ASSEMBLY) {
      for (AssemblyType assemblyType : AssemblyType.values()) {    
        validationPlan.getProperty().getOptions().assemblyType = assemblyType; 
        ValidationResult result = validationPlan.execute(entry);
        assertTrue(result.isValid()); // No sequence name length validation
        }
      }
   }
   else {
       ValidationResult result = validationPlan.execute(entry);
       assertEquals(!expectError, result.isValid());
   }
```